### PR TITLE
feat: make assert function generation parametrized

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/2" target="_blank">Graphite</a> 
      * **PR #3** <a href="https://app.graphite.dev/github/pr/skyleague/therefore/3" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/3?utm_source=stack-comment).

File: examples/json/json.type.ts [hunk 1]
```diff
@@ -24,11 +24,6 @@ export const Json = {
     source: `${__dirname}json.schema`,
```
Comment (MickVanDuijn) [inline examples/json/json.type.ts:24]: <i class='graphite__hidden'>[Re: line 6]<br/></i>

This import is still generated<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/skyleague/therefore/3?utm_source=unchanged-line-comment">Graphite</a>.</span>

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
See inline comments

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/therefore/3?utm_source=stack-comment).


Job Reference: EZc1hcjGE8w9A7wKUxQS

**Pull request merged**